src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParameterInput.js

```diff
@@ -1,17 +1,32 @@
 // Copyright (c) Cosmo Tech.
 // Licensed under the MIT license.
 
-import React from 'react';
+import React, { useCallback, useRef } from 'react';
 import { ConfigUtils } from '../../../../utils/ConfigUtils';
 import { VAR_TYPES_COMPONENTS_MAPPING } from '../../../../utils/scenarioParameters/VarTypesComponentsMapping';
 import { useTranslation } from 'react-i18next';
 import PropTypes from 'prop-types';
+import { useStore } from 'react-redux';
 
 const ScenarioParameterInput = ({ parameterData, parametersState, setParametersState, context }) => {
   const { t } = useTranslation();
   const parameterVarType = ConfigUtils.buildExtendedVarType(parameterData.varType, parameterData.subType);
   let varTypeFactory;
 
+  const store = useStore();
+
+  const currentScenarioId = useCallback(() => store.getState().scenario?.current?.data?.id, [store]);
```

Comment:
I don't know if there is a different convention when using `useCallback`, but do you think we could rename `currentScenarioId` to `getCurrentScenarioId` to show that it is a function ?

Reply:
Right ;)